aws/resource_aws_ec2_client_vpn_authorization_rule_test.go

```diff
@@ -0,0 +1,326 @@
+				),
+			},
+		},
+	})
+}
+
+func testAccAwsEc2ClientVpnAuthorizationRule_disappears(t *testing.T) {
+	var v ec2.AuthorizationRule
+	rStr := acctest.RandString(5)
+	resourceName := "aws_ec2_client_vpn_authorization_rule.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckClientVPNSyncronize(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsEc2ClientVpnAuthorizationRuleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEc2ClientVpnAuthorizationRuleConfigBasic(rStr),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsEc2ClientVpnAuthorizationRuleExists(resourceName, &v),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsEc2ClientVpnAuthorizationRule(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsEc2ClientVpnAuthorizationRuleDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).ec2conn
+
+	defer testAccEc2ClientVpnEndpointSemaphore.Notify()
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_ec2_client_vpn_authorization_rule" {
+			continue
+		}
+
+		_, err := findClientVpnAuthorizationRuleByID(conn, rs.Primary.ID)
+		if err == nil {
+			return fmt.Errorf("Client VPN authorization rule (%s) still exists", rs.Primary.ID)
+		}
+		if isAWSErr(err, tfec2.ErrCodeClientVpnAuthorizationRuleNotFound, "") || isAWSErr(err, tfec2.ErrCodeClientVpnEndpointIdNotFound, "") {
+			continue
+		}
+		return err
+	}
+
+	return nil
+}
+
+func testAccCheckAwsEc2ClientVpnAuthorizationRuleExists(name string, assoc *ec2.AuthorizationRule) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).ec2conn
+
+		result, err := findClientVpnAuthorizationRuleByID(conn, rs.Primary.ID)
+		if err != nil {
+			return fmt.Errorf("error reading Client VPN authorization rule (%s): %w", rs.Primary.ID, err)
+		}
+
+		if result != nil || len(result.AuthorizationRules) == 1 || result.AuthorizationRules[0] != nil {
+			*assoc = *result.AuthorizationRules[0]
+			return nil
+		}
+
+		return fmt.Errorf("Client VPN network association (%s) not found", rs.Primary.ID)
+	}
+}
+
+func testAccEc2ClientVpnAuthorizationRuleConfigBasic(rName string) string {
+	return testAccEc2ClientVpnVpcComposeConfig(rName, 1, `
+resource "aws_ec2_client_vpn_authorization_rule" "test" {
+  client_vpn_endpoint_id = aws_ec2_client_vpn_endpoint.test.id
+  target_network_cidr    = aws_subnet.test[0].cidr_block
+  authorize_all_groups   = true
+}
+`)
+}
+
+func testAccEc2ClientVpnAuthorizationRuleConfigGroups(rName string, groupNames map[string]string) string {
+	var b strings.Builder
+	for k, v := range groupNames {
+		fmt.Fprintf(&b, `
```

Comment:
Historically we have preferred to build out separate, verbose configurations so it's easier for those less/unfamiliar with Go or the testing code to understand the final configuration being built. The composition pattern we have been using lately is more centered around creating modular or base chunks of testing configuration, that are hopefully still very readable or possibly copy-paste friendly into a real configuration. We may want to consider the implications of string builders, etc. for newcomers, even if it increases the line count in the end by writing out more verbose configurations.